In [3]:
from Bio import SeqIO

In [4]:
# There's only one seqence record in the NC_000913_3.gb file
for refseq in SeqIO.parse("./NC_000913_3.gb", "genbank"):
    print(refseq.id)
    print(repr(refseq.seq))
    

glpR_feat = None
for f in refseq.features:
    if (f.type == "gene") and ("glpR" in f.qualifiers["gene"]):
        display(f.qualifiers)
        glpR_feat = f
        break
glpR_feat


glpR_seq = glpR_feat.location.extract(refseq).reverse_complement().seq  # glpR is on the reverse strand
display(len(glpR_seq), glpR_feat.location, glpR_seq)

glpR_seq_posve_strand = glpR_feat.location.extract(refseq).seq
print()
print(glpR_seq_posve_strand)
NC_glpR = glpR_seq_posve_strand

NC_000913.3
Seq('AGCTTTTCATTCTGACTGCAACGGGCAATATGTCTCTGTGTGGATTAAAAAAAG...TTC', IUPACAmbiguousDNA())


OrderedDict([('gene', ['glpR']),
             ('locus_tag', ['b3423']),
             ('gene_synonym', ['ECK3409; JW3386']),
             ('pseudo', ['']),
             ('db_xref', ['EcoGene:EG10400', 'GeneID:947926'])])

758

FeatureLocation(ExactPosition(3559847), ExactPosition(3560605), strand=-1)

Seq('TCAGCACAGCTCCAGTTGAATATGGTGGTCCGTCAGCACCTGCATCACGCTTAC...CAT', IUPACAmbiguousDNA())


ATGAAACAAACACAACGTCACAACGGTATTATCGAACTGGTTAAACAGCAGGGTTATGTCAGTACCGAAGAGCTGGTAGAGCATTTCTCCGTCAGCCCGCAGACTATTCGCCGCGACCTCAATGAGCTGGCGGAGCAAAACCTGATCCTGGCCATCATGGCGGTGCGGCGCTGCCTTCCAGTTCGGTTAACACGCCGTGGCACGATCGCAAGGCCACCCAGACCGAAGAAAAAGAGCGCATCGCCCGCAAAGTGGCGGAGCAAATCCCCAATGGCTCGACGCTGTTTATCGATATCGGCACCACGCCGGAAGCGGTAGCGCACGCACTGCTCAATCACAGCAATTTGCGCATTGTCACCAACAATCTCAACGTTGCTAACACGTTGATGGTAAAAGAAGATTTTCGCATCATTCTCGCCGGTGGCGAATTACGCAGCCGCGATGGCGGGATCATTGGCGAAGCGACGCTCGATTTTATCTCCCAGTTCCGCCTTGATTTCGGCATTCTGGGGATAAGCGGCATCGATAGCGACGGCTCGCTGCTGGAGTTCGATTACCACGAAGTTCGCACCAAACGCGCCATTATTGAGAACTCGCGCCACGTTATGCTGGTTGTCGATCACTCGAAATTTGGCCGTAACGCGATGGTCAATATGGGCAGCATCAGCATGGTAGATGCCGTCTACACCGACGCCCCGCCGCCAGTAAGCGTGATGCAGGTGCTGACGGACCACCATATTCAACTGGAGCTGTGCTGA


In [5]:
for refseq in SeqIO.parse("./U00096.2.gb", "genbank"):
    print(refseq.id)
    print(repr(refseq.seq))
    

glpR_feat = None
for f in refseq.features:
    if (f.type == "gene") and ("glpR" in f.qualifiers["gene"]):
        display(f.qualifiers)
        glpR_feat = f
        break
glpR_feat


glpR_seq = glpR_feat.location.extract(refseq).reverse_complement().seq  # glpR is on the reverse strand
print(len(glpR_seq), glpR_seq)

glpR_seq_posve_strand = glpR_feat.location.extract(refseq).seq
print()
print(glpR_seq_posve_strand)
U_glpR = glpR_seq_posve_strand

U00096.2
Seq('AGCTTTTCATTCTGACTGCAACGGGCAATATGTCTCTGTGTGGATTAAAAAAAG...TTC', IUPACAmbiguousDNA())


OrderedDict([('gene', ['glpR']),
             ('locus_tag', ['b3423']),
             ('gene_synonym', ['ECK3409', 'JW3386']),
             ('db_xref', ['EcoGene:EG10400'])])

759 TCAGCACAGCTCCAGTTGAATATGGTGGTCCGTCAGCACCTGCATCACGCTTACTGGCGGCGGGGCGTCGGTGTAGACGGCATCTACCATGCTGATGCTGCCCATATTGACCATCGCGTTACGGCCAAATTTCGAGTGATCGACAACCAGCATAACGTGGCGCGAGTTCTCAATAATGGCGCGTTTGGTGCGAACTTCGTGGTAATCGAACTCCAGCAGCGAGCCGTCGCTATCGATGCCGCTTATCCCCAGAATGCCGAAATCAAGGCGGAACTGGGAGATAAAATCGAGCGTCGCTTCGCCAATGATCCCGCCATCGCGGCTGCGTAATTCGCCACCGGCGAGAATGATGCGAAAATCTTCTTTTACCATCAACGTGTTAGCAACGTTGAGATTGTTGGTGACAATGCGCAAATTGCTGTGATTGAGCAGTGCGTGCGCTACCGCTTCCGGCGTGGTGCCGATATCGATAAACAGCGTCGAGCCATTGGGGATTTGCTCCGCCACTTTGCGGGCGATGCGCTCTTTTTCTTCGGTCTGGGTGGCCTTGCGATCGTGCCACGGCGTGTTAACCGAACTGGAAGGCAGCGCCGCACCGCCATGATGGCGCAGGATCAGGTTTTGCTCCGCCAGCTCATTGAGGTCGCGGCGAATAGTCTGCGGGCTGACGGAGAAATGCTCTACCAGCTCTTCGGTACTGACATAACCCTGCTGTTTAACCAGTTCGATAATACCGTTGTGACGTTGTGTTTGTTTCAT

ATGAAACAAACACAACGTCACAACGGTATTATCGAACTGGTTAAACAGCAGGGTTATGTCAGTACCGAAGAGCTGGTAGAGCATTTCTCCGTCAGCCCGCAGACTATTCGCCGCGACCTCAATGAGCTGGCGGAGCAAAACCTGATCCTGCGCCATCATGGCGGTGCGGCGCTGCCTTCCAGTTCGGTTAACACGCCGTGGCACGATCGCAAGGCCACCCAGACCGAAGAAAAAG

The only difference between the NC_000913.3 and U00096.2 glpR sequences (reverse complement since on the negative strand) is a G, similar what the BreSeq report mutations had relative to the NC_000913.3 genome reference. This difference is additionally demonstrated in the genes lengths shown in the results above (758 for NC_000913.3 vs 759 U00096.2, a difference of 1).  
The relative position of the difference on the glpR positive strand is 608 (1-based positioning; used Meld to find this).  
The positive strand start position for glpR on NC_000913.3 is 3559847.  
3559847 + 608 = 3560455  
3560455 is the position of the insertion in the BOP27 strain.  
Our BOP27 strain therefore has a functional glpR gene.  